mongo/single_result.go

```diff
@@ -29,20 +28,25 @@ type SingleResult struct {
 	reg *bsoncodec.Registry
 }
 
-// NewSingleResultFromBytes creates a SingleResult with an underlying Cursor pre-loaded with the provided contents.
-func NewSingleResultFromBytes(contents []byte) *SingleResult {
-	c := &Cursor{
-		bc:       driver.NewBatchCursorFromBytes(contents),
-		registry: bson.DefaultRegistry,
-		Current:  bson.Raw(contents),
+// NewMockSingleResult creates a SingleResult with the provided error, registry, and an underlying Cursor pre-loaded with
```

Comment:
I suggest removing `Mock` from the name in case there are future use-cases. Perhaps this could be `NewSingleResultFromDocument`?

Reply:
Changed to `NewSingleResultFromDocument`.